src/QsFmt/SPECIFICATION.md

```diff
@@ -0,0 +1,58 @@
+These two commands work separately and can't be run together. If the user wishes to both format and update their code, they would need to run the tool twice, first with 'update' and then with 'format'.
+
+The tool will initially only support the 'update' command, as the formatting functionality is still being worked on. The 'format' command will be unavailable until it is supported.
+
+## Input and Output
+Input files to the formatter will be the last argument(s) to the tool. Zero to many arguments may be given.
+ - The default behavior, which is used if no input arguments are provided, is for the formatter to take as input all files with the '.qs' extension in the current working directory.
+ - Inputs can be given as one to many paths to directories, in which all files with the '.qs' extension found will be taken as input.
+ - Inputs can be given as one to many paths to individual files with the '.qs' extension.
+ - Inputs can be any combination of paths to files and paths to directories.
+ - The `--recurse` option can be specified to process the input directories recursively for input files. You can specify this for all input directories, not per-directory.
+
+The output of the formatter is to overwrite the input files that it processes.
+
+## Rules that are run
+
+The tool will parse the input into a concrete syntax tree using ANTLR. This tree will then have rules applied to it in the form of transformations on the tree.
+The transformations are separated into two groups: transformations run by the 'format' command, and transformations run by the 'update' command.
+Listed here are some of the transformations we intend to use.
+
+### Formatting Transformations
+
+Collapsed Spaces - removes duplicate spaces so that there is only a single space where spaces are used
+Operator Spacing - ensure that operators have the appropriate spaces in them
+New Lines - removes excessive new lines
+Indentation - ensures that there are appropriate indentation for the text
+
+### Updating Transformations
+
+Updating Transformations remove outdated syntax that is deprecated and will be no longer supported in the future.
+
+Array Syntax - [proposal](https://github.com/microsoft/qsharp-language/blob/main/Approved/2-enhanced-array-literals.md)
+Using Syntax - [proposal](https://github.com/microsoft/qsharp-language/blob/main/Approved/1-implicitly-scoped-qubit-allocation.md)
+
+## How errors are handled
+
+If an exception is thrown, that will be surfaced to the user and is a bug.
+
+Parsing errors may occur if the formatter is given text that is not proper Q# code. In this case ANTLR gives the appropriate error message.
+Errors should not occur during a formatting transformation. All formatting transformations should never encounter the syntax they are expected to change and be unable to perform the change.
```

Comment:
I suspect you meant to write
> All formatting transformations should never encounter **an error in** the syntax they are expected to change

Reply:
Hmm, not quite. I'm not sure the best way to word this, but the formatting transformations should not (see the syntax they need to change AND not be able to change it). If they see the syntax they are targeting, they should always be able to perform their change.

Reply:
Then I assume that the sentence means the following:
> All formatting transformations should never encounter the syntax **_that_** they are expected to change **_but are unable to_** change

Reply:
applied